src/request/selectors/RequestDetailRequestSelectors.ts

```diff
@@ -1,3 +1,47 @@
 import { IRequestState } from '../stores/IRequestState';
+import { IRequestDetailRequestState } from '../stores/IRequestDetailRequestState';
+import { IRequestDetailRequestMiddlewareState } from '../stores/IRequestDetailRequestMiddlewareState';
 
-export const getRequest = (state: IRequestState) => state.detail.request;
+import { createSelector } from 'reselect';
+
+const getMiddlewareState = (state: IRequestState) => state.detail.request.middleware;
+
+export const getUrl = (state: IRequestState) => state.detail.request.url;
+export const getRequest = (state: IRequestState) => state.detail.request.request;
+export const getResponse = (state: IRequestState) => state.detail.request.response;
+
+interface IFlattenedMiddleware {
```

Comment:
Do we want to come up with a general name for these "first class" "view models"... like `IMiddlewareViewModel`? I don't think that this needs to be done now and can wait till we get over all the tabs into the strawman. I'm sure that at this point we will see other patterns we want to bring across as well.